kliff/transforms/configuration_transforms/graphs/generate_graph.py

```
@@ -0,0 +1,199 @@
+    For porting the graph representation you also need to provide the TorchScript model file name.
+
+    Args:
+        species (list): List of species.
+        cutoff (float): Cutoff distance.
+        n_layers (int): Number of convolution layers.
+        as_torch_geometric_data (bool): If True, the graph will be returned as a Pytorch
+            Geometric Data object.
+        implicit_fingerprint_copying (bool): If True, the fingerprint will be copied to
+            the Configuration object's fingerprint attribute.
+    """
+
+    def __init__(
+        self,
+        species,
+        cutoff,
+        n_layers,
+        as_torch_geometric_data=False,
+        implicit_fingerprint_copying=False,
+    ):
+        super().__init__(implicit_fingerprint_copying=implicit_fingerprint_copying)
+        self.species = species
+        self.cutoff = cutoff
+        self.n_layers = n_layers
+        self.infl_dist = n_layers * cutoff
+        self._tg = graph_module
+        if as_torch_geometric_data and not torch_geometric_available():
+            raise ImportError("Torch geometric is not available")
+        self.as_torch_geometric_data = as_torch_geometric_data
+
+    def forward(self, configuration: "Configuration"):
+        """
+        Generate a graph representation of a configuration.
+
+        Args:
+            configuration: Instance of ~:class:`kliff.dataset.Configuration`. For which the
+                graph representation is to be generated.
+
+        Returns:
+            C++ custom graph object or Pytorch Geometric Data object.
+        """
+        graph = graph_module.get_complete_graph(
+            self.n_layers,
+            self.cutoff,
+            configuration.species,
+            configuration.coords,
+            configuration.cell,
+            configuration.PBC,
+        )
+        graph.energy = configuration.energy
+        graph.forces = configuration.forces
+        if torch_available:
```

This `if torch_available` is error prone, for example, in the case a user wants to export to PyG graph, but forget to install torch.

Essentially, this should only be detemined by `self.as_torch_geometric_data`. If it is true, we should export a PyG graph. We can raise error if torch is not available.

---

See above. Now `PyGGraph` is always used. If PyG is present then it will be an instance of `pyg.data.Data` otherwise it will be a plain Python class with same fields. In the latter case it will log warning as well.